plugins/out_s3/s3.c

```diff
@@ -1129,7 +1200,6 @@ static int upload_data(struct flb_s3 *ctx, struct s3_file *chunk,
                 return -1;
             }
         }
-        return FLB_RETRY;
```

Comment:
@Claych See the code above this line where we decrement the index number if the request failed. You change has made it so the code will return before this line.

We need to make sure we don't change this existing feature. 

Please fix this and then try testing with with the `$INDEX` feature in S3 key noted in the docs: https://docs.fluentbit.io/manual/pipeline/outputs/s3